tasks/ansible.yml

```diff
@@ -22,7 +22,10 @@
     src: /root/.ssh/id_rsa.pub
   register: ansible_public_key_slurp
 
-- name: Save SSH public key to dummy host
-  add_host:
-    name: OPENSHIFT_ANSIBLE_HOST
+- name: Save SSH public key to fact
+  set_fact:
     public_key: "{{ ansible_public_key_slurp.content | b64decode }}"
+
+- name: Create group of Ansible hosts
+  group_by:
+    key: openshift_ansible_hosts
```

Comment:
I don't think is how the module works: https://docs.ansible.com/ansible/latest/modules/group_by_module.html

Also, `openshift_ansible_hosts` is undefined.

You probably want to group on `{{ openshift_is_ansible_host }}`

Reply:
Pretty sure that's how it works.

Every host this play is run for is added to the group ' openshift_ansible_hosts'. This is a string, not a variable. The play (all of ansible.yml) is only run for hosts for which `openshift_is_ansible_host` is true.

Reply:
Tested it.

```
- name: Create group of Ansible hosts
  group_by:
    key: openshift_ansible_hosts

- debug:
    msg: "{{groups}}"
```

Output in my setup (shortened):

```
ok: [master1.oc.arestless.com] => {
    "msg": {
        "all": [
            "git.arestless.com", 
            "localhost", 
            "master1.oc.arestless.com"
        ],
        "ansible": [
            "localhost"
        ],
        "git": [
            "git.arestless.com"
        ], 
        "openshift_ansible_hosts": [
            "master1.oc.arestless.com"
        ], 
        "openshift_cluster": [
            "master1.oc.arestless.com"
        ],
        "ungrouped": [],
    }
}
```

Reply:
Makes sense, I was wrong